app/gui2/src/util/array.ts

```diff
@@ -6,3 +6,23 @@ export function findIndexOpt<T>(arr: T[], pred: (elem: T) => boolean): number |
   const index = arr.findIndex(pred)
   return index >= 0 ? index : null
 }
+
+/** Binary search, using a predicate `pred` that returns `false` for elements ordered before the
+ * target, and `true` for elements ordered after the target. Returns the index of the first element
+ * for which the predicate returns `true`.
```

Comment:
The doc is a bit inconsistent about the `true` case. The first sentence alone made me stop and think, what should I return for the `target` itself? Also, what really is `target` here? What happens if a `target` I have in mind is not actually in the searched array?

That's because this is not really a "binary search" for a specific element. Instead, it is a search of an array partition index. Therefore I believe it should be renamed. See Rust docs for [`partition_point`](https://doc.rust-lang.org/std/primitive.slice.html#method.partition_point) as a reference, and contrast that with [`binary_search` docs](https://doc.rust-lang.org/std/primitive.slice.html#method.binary_search_by), which does an actual search for specific element using a comparator and has a failure case.